extensions-utils/src/dfx.rs

```diff
@@ -50,6 +54,57 @@ where
         })
 }
 
+/// Calls a binary that was delivered with an extension tarball.
+///
+/// # Returns
+/// - On success, returns stdout as a string.
+/// - On error, returns an error message including stdout and stderr.
+#[context("Calling {} CLI failed, or, it returned an error.", binary_name)]
+pub fn call_extension_bundled_binary<S, I>(binary_name: &str, args: I) -> anyhow::Result<String>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let extension_binary_path =
+        std::env::current_exe().map_err(|e| anyhow::anyhow!("Failed to get current exe: {}", e))?;
+    let extension_dir_path = extension_binary_path
+        .parent()
+        .unwrap_or_else(|| Path::new("."));
```

Comment:
Can we treat this as an error?  Is there a circumstance in which we would want to actually continue here?